packages/babel-plugin/src/index.js

```diff
@@ -12,6 +12,20 @@ import coreMacro, {
 } from './core-macro'
 import { getStyledOptions, createTransformerMacro } from './utils'
 
+const EMOTION_BABEL_PLUGIN_ANNOTATION_REGEX = /\*?\s*@emotion\/babel-plugin\s+(.+$)/
+
+// this is really intended to only be used internally
+const getEmotionBabelPluginPragmaOptions = state => {
+  if (!state.file.ast.comments) {
+    return {}
+  }
+  return state.file.ast.comments
+    .map(comment => comment.value.match(EMOTION_BABEL_PLUGIN_ANNOTATION_REGEX))
+    .filter(Boolean)
+    .map(match => JSON.parse(`{${match[1]}}`))
+    .reduce((a, b) => Object.assign(a, b), {})
+}
```

Comment:
I'd rather we didn't have this because people could start relying on it. Maybe we could do the same parsing in the babel test infra?

Reply:
We could - that was my initial intention but I didn't have time to figure out how this should be done properly. Do you have any idea how this could be done without sharing the parsed options through the state (which would basically also allow people to abuse it)? I didn't want to set up any more complicated infra that would sidestep Babel for this.

In the ideal situation, we'd just provide those options through regular options but that can't be done easily to the best of my knowledge (unless we read files in `test` and do our own parsing there).